rgw/admin/quota.go

```diff
@@ -0,0 +1,56 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// Quota describes an object store quota for a user or a bucket
+type Quota struct {
+	UID        string `json:"user_id" url:"uid"`
+	QuotaType  string `url:"quota-type"`
+	Enabled    *bool  `json:"enabled" url:"quota,enabled"`
+	CheckOnRaw bool   `json:"check_on_raw"`
+	MaxSize    *int64 `json:"max_size" url:"max-size"`
+	MaxSizeKb  *int64 `json:"max_size_kb" url:"max-size-kb"`
+	MaxObjects *int64 `json:"max_objects" url:"max-objects"`
+}
+
+// GetQuota will return the quota for a bucket or a user
+func (api *API) GetQuota(ctx context.Context, quota Quota) (*Quota, error) {
```

Comment:
I note that this function returns a Quota pointer but this function and the SetQuota function take a non-pointer Quota. I hadn't noticed this when looking at the earlier PR, and so I do see it is consistent with the other functions in rgw/admin. It just strikes me a bit quirky. Not a big deal though.

Reply:
I agree with that.

Reply:
@leseb could you react to this, so we know you didn’t miss it?

Reply:
Ah yes, they are always pros and cons, and the easy pros is that we can return `nil` in case of an error instead of an empty struct. Also, selectors automatically dereference pointers to structs so it's implied when accessing values and not extra work for the caller.

I often read that we should use pointers if the returned object's content can/should be changed, if not let's not use a pointer.
The struct types we return are not meant to be modified, they are what are.

If you have more arguments and feel strongly about it I can change and stop returning pointers.

@phlogistonjohn @ansiwen thoughts?

Reply:
My rule is usually: value character -> no pointer, identity character (that is, should not be duplicated usually) -> pointer. So here I personally would go with no pointer.
But I think our concern here is mainly consistency. So that id you chose a pointer as return value, the argument of the same type should also be a pointer then.

Reply:
Sounds good, I'll go ahead and remove the pointers then.